src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetWidget.java

```diff
@@ -575,6 +576,23 @@ public void run()
       toolbar.addRightSeparator();
       toolbar.addRightWidget(toggleDocOutlineButton_);
       
+      showWhitespaceCharactersCheckbox_ = new CheckBox("Show whitespace characters");
+      showWhitespaceCharactersCheckbox_.setVisible(false);
+      showWhitespaceCharactersCheckbox_.setValue(uiPrefs_.showInvisibles().getValue());
+      showWhitespaceCharactersCheckbox_.addValueChangeHandler((ValueChangeEvent<Boolean> event) -> {
+         editor_.setShowInvisibles(event.getValue());
+      });
+      
+      if (docUpdateSentinel_ != null && docUpdateSentinel_.getPath() != null)
+      {
+         FileSystemItem item = FileSystemItem.createFile(docUpdateSentinel_.getPath());
+         String ext = item.getExtension();
+         if (".csv".contentEquals(ext) || ".tsv".equals(ext))
```

Comment:
why `contentEquals` vs `equals`?

Reply:
Because my Eclipse autocomplete prefers to insert `contentEquals()` rather than `equals()` when I type `eq<TAB>`. Oops 😬